CPP-Examples.cpp

```diff
@@ -11,16 +11,17 @@ int main()
 
    strcpy(dest, src); 
  
-   strncpy(dest,src, strlen(dest)); //DevSkim: ignore DS111237,DS154189 
+   strncpy(dest,src, strlen(dest)); 
    
-   strcat(dest, "this is bad");
+   strcat(dest, "this is bad");     //DevSkim: ignore all
 
    strncat(dest, "this is also bad", strlen(dest)); 
 
    memcpy(dest2, dest, sizeof dest);
```

Comment:
[Problematic C function detected (memcpy)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS121708.md#DS121708)
There are a number of conditions in which memcpy can introduce a vulnerability (mismatched buffer sizes, null pointers, etc.). More secure alternitives perform additional validation of the source and destination buffer

*Fix Guidance:* Use memcpy_s if possible.


Suggestions:
[1] memcpy_s(dest2, <size of dest2>,  dest,  sizeof dest);